### PR TITLE
Release process: automatically set milestone date after a successful release

### PR DESCRIPTION
**Pull request by tarunvijwani**
This PR adds a new automation that will update the milestone due date after the release. 

## Changes
- Added a new automation in `lib\automations\update-milestone` folder. 
- Updated the `action.yml` file to read the target milestone number. 

# Testing
To test this, you will need to:
- **Fork** the `woocommerce-gutenberg-products-block` and then unfork it (Go to this url: https://support.github.com/contact?tags=rr-forks and type `unfork` into the subject line. Follow the steps from the virtual assistant and wait 5-10 mins for the request to be actioned).
- Enable issues on your new unforked repository (go to `Settings > General > Features > Issues`)
- Activate GH actions
- Give read and write permissions to the workflows: Settings > Actions > General > Workflow permissions
- Allow GH actions to create PRs: Settings > Actions > General
- Create a milestone (Example: 11.1.0) and close the milestone
- Replace your [wordpress-deploy.yml](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/.github/workflows/wordpress-deploy.yml) file with [wordpress-deploy.yml](https://github.com/tarunvijwani/woocommerce-blocks/blob/trunk/.github/workflows/wordpress-deploy.yml). It will remove the deployment steps. 
- Create a new milestone say 11.0.0 without any due date and close it. 
- Draft a new release from https://github.com/tarunvijwani/woocommerce-blocks/releases/new
- Give the milestone number in name and tag.
- Publish the release. 
- Go to actions and wait for [wordpress-deploy.yml](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/.github/workflows/wordpress-deploy.yml) to finish.
- If it fails, check the debug logs. (It might get fail because we don't have any PR assigned to the milestone for the release)
- Check and confirm milestone is updated with the due date.
- Please test around this.

### CODE REVIEW

**Comment by mikejolley** [comment]
@tarunvijwani do you have a forked repo where you tested that? If you're ran through the above steps I'm happy to trust that, since a failure here would be low impact.

File: action.yml
Comment (mikejolley) [inline action.yml]: This description is pretty hard to understand. Can you elaborate what `for modifying any milestone.` means?

File: lib/automations/update-milestone/get-config.js
Comment (mikejolley) [inline lib/automations/update-milestone/get-config.js]: Are there any dangers to outputting the user input?

File: lib/automations/update-milestone/update-milestone-handler.js
Comment (mikejolley) [inline lib/automations/update-milestone/update-milestone-handler.js]:
```suggestion
			`Update Milestone: Target milestone: ${ config.targetMilestone } already has a due date.`
```

File: lib/automations/update-milestone/update-milestone-handler.js
Comment (mikejolley) [inline lib/automations/update-milestone/update-milestone-handler.js]:
```suggestion
		`Update Milestone: Milestone ${ config.targetMilestone } successfully updated with the ${ dueDate } due date.`
```

File: action.yml
Comment (mikejolley) [reply action.yml]: If I'm understanding correctly, how about: 
```suggestion
            Title of the milestone that will be updated.
```

**Review by mikejolley — APPROVED** [review]
Code looks solid enough - I made some suggestions for text. I have **not** tested this, but the steps look valid. If you've tested this on a fork with success, I'm happy for this to be published. Thanks!

File: action.yml
Comment (tarunvijwani) [reply action.yml]: Yes, that's the perfect description. Thank you!

File: lib/automations/update-milestone/get-config.js
Comment (tarunvijwani) [reply lib/automations/update-milestone/get-config.js]: @mikejolley That's a very good point. I'm not sure about it but I would like to learn the consequence of it. I'll appreciate it if you can share more details. 

I'll commit the below change for now: 
```suggestion
			`Update Milestone: Missing required input ${ input.input }`
```

**Comment by tarunvijwani** [comment]
>@tarunvijwani do you have a forked repo where you tested that? If you're ran through the above steps I'm happy to trust that, since a failure here would be low impact.

Yes, I have tested it in my repo: https://github.com/tarunvijwani/woocommerce-blocks

Thank you for reviewing the changes. I'll merge it once the test cases are ready. 👍

**Comment by tarunvijwani** [comment]
Test cases for the automation are added in the PR: https://github.com/woocommerce/automations/pull/57

Once the test case PR is merged, I'll test it again and merge it.

**Comment by tarunvijwani** [comment]
Documentation of the automation are added in the PR: https://github.com/woocommerce/automations/pull/58

**Pull request merged**
